library/std/src/primitive_docs.rs

```diff
@@ -11,8 +11,9 @@
 /// `bool` implements various traits, such as [`BitAnd`], [`BitOr`], [`Not`], etc.,
 /// which allow us to perform boolean operations using `&`, `|` and `!`.
 ///
-/// `if` always demands a `bool` value. [`assert!`], which is an important macro in testing,
-/// checks whether an expression returns `true` and panics if it isn't.
+/// `if` always demands a `bool` value as its conditional. [`assert!`], which is an
```

Comment:
Maybe

```suggestion
/// `if` requires a `bool` value as its conditional. [`assert!`], which is an
```

would be better wording?